### PR TITLE
5383 Update error messages bulk qts

### PR DESCRIPTION
**Pull request by richardpattinson**
### Context

Update error messages for bulk qts uploads. See trello cards for spreadsheet containing changes.

### CODE REVIEW

**Comment by AbigailMcP** [comment]
[Update error messages for bulk recommendation](https://trello.com/c/AEtEZFNS/5383-update-error-messages-for-bulk-recommendation)

File: config/locales/en.yml [hunk 1]
```diff
@@ -1320,40 +1320,40 @@ en:
...
-              last_names: Trainee last names do not match
-              lead_school: Lead school does not match
-              qts_or_eyts: QTS/EYTS declaration does not match
-              route: Route does not match
-              phase: Phase does not match
-              age_range: Age range does not match
-              subject: Subject does not match
+              trn_format: TRN must be 7 numbers
+              hesa_id_format: HESA ID must be 13 or 17 numbers
+              award_date_future: Date QTS or EYTS standards met must be today or in the past
```
Comment (edwardhorsford) [inline config/locales/en.yml:1342]: @johndoates didn't we talk about saying the date cannot be in the future?

**Review by AbigailMcP — COMMENTED** [review]
Do the tests pass? I think some of these strings will be referenced in the spec files so these will need updating.

File: app/services/bulk_update/recommendations_uploads/validate_trainee.rb [hunk 2]
```diff
@@ -84,6 +84,14 @@ def trainees_by_hesa_id!
         end
       end
 
+      def id_columns
+        output = []
+        %i[trn hesa_id provider_trainee_id].each do |field|
+          output >> field if row.send(field)
```
Comment (jhenderson) [inline app/services/bulk_update/recommendations_uploads/validate_trainee.rb:90]:
```suggestion
          output << field if row.send(field)
```

File: app/services/bulk_update/recommendations_uploads/validate_trainee.rb [hunk 3]
```diff
@@ -36,13 +36,13 @@ def find!
...
-        return @messages << error_message(:multiple_trainees_trn_recieved, found_with:) if trainees.size > 1
+        return @messages << error_message(:multiple_trainees_trn_received, count: trainees.count, found_with: found_with) if trainees.size > 1
         # otherwise check if multiple in non-draft non-trn-received were found
-        return @messages << error_message(:multiple_trainees_found_via, found_with:) if trainees_not_trn_received.size > 1
+        return @messages << error_message(:multiple_trainees_found_via, count: trainees.count, found_with: found_with) if trainees_not_trn_received.size > 1
         # otherwise check if one was found outside of non-draft, non-trn_received
         return @messages << error_message(:trainee_wrong_state, state: trainee_not_trn_received.state.humanize) if trainee_not_trn_received
 
-        @messages << error_message(:no_trainee_matched)
+        @messages << error_message(:no_trainee_matched, id_available: id_columns)
```
Comment (jhenderson) [inline app/services/bulk_update/recommendations_uploads/validate_trainee.rb:45]: Can we add a test example to cover the added information?

**Review by jhenderson — CHANGES_REQUESTED** [review]

File: app/services/bulk_update/recommendations_uploads/validate_trainee.rb (hunk 3 above)
Comment (richardpattinson) [reply app/services/bulk_update/recommendations_uploads/validate_trainee.rb:45]: Should be in there now

**Comment by richardpattinson** [comment]
[Update error messages for bulk recommendation](https://trello.com/c/AEtEZFNS/5383-update-error-messages-for-bulk-recommendation)

File: config/locales/en.yml (hunk 1 above)
Comment (johndoates) [reply config/locales/en.yml:1349]: We did but I've used the standard wording from the [design system](https://design-system.service.gov.uk/components/date-input/#error-messages).

**Comment by johndoates** [comment]
I can confirm that the wording of all the error messages has been updated correctly.

There are a couple of error messages with variables where the capitalisation is not correct.

1. trainee_wrong_state uses the %{state} variable. This should have one of the following values, with the capitalisation as written here because it's appearing in the middle of a sentence:

- deferred
- pending TRN
- EYTS awarded
- QTS awarded
- EYTS recommended
- QTS recommended
- withdrawn

Let me know if I've missed any of the statuses here.

2.  no_trainee_matched uses the %{id_available} variable. This should have one of the following values, with the capitalisation as written here because it's appearing at the start of a sentence:

- TRN
- Provider trainee ID
- HESA ID
- TRN or provider trainee ID
- TRN or HESA ID
- Provider trainee ID or HESA ID
- TRN, provider trainee ID or HESA ID
 
3. multiple_trainees_trn_recieved and multiple_trainees_found_via use the %{found_with} variable. I wasn't able to check these easily so they may already be correct, but the variable should have one of the following values, with the capitalisation as written here:

- TRN
- provider trainee ID
- HESA ID

Since I was unable to easily check multiple_trainees_trn_recieved and multiple_trainees_found_via, I couldn't check that %{id_available} appeared correctly in those error messages.

**Comment by johndoates** [comment]
I tested a few variations of no_trainee_matched and the capitalisation now looks fine to me.

However, when I tried trainee_wrong_state the capitalisation was incorrect. For example I got:

* The trainee has Deferred status - you can only recommend trainees with TRN received status
* The trainee has Recommended for award status - you can only recommend trainees with TRN received status

These should be 'deferred' and 'recommended'. Could you please check the capitalisation of the statuses?

As before, I couldn't check multiple_trainees_trn_recieved and multiple_trainees_found_via. I don't think that I can create a trainee with a duplicated ID, since the Register interface prevents me from doing so.

**Comment by richardpattinson** [comment]
> I tested a few variations of no_trainee_matched and the capitalisation now looks fine to me.
> 
> However, when I tried trainee_wrong_state the capitalisation was incorrect. For example I got:
> 
> * The trainee has Deferred status - you can only recommend trainees with TRN received status
> * The trainee has Recommended for award status - you can only recommend trainees with TRN received status
> 
> These should be 'deferred' and 'recommended'. Could you please check the capitalisation of the statuses?
> 
> As before, I couldn't check multiple_trainees_trn_recieved and multiple_trainees_found_via. I don't think that I can create a trainee with a duplicated ID, since the Register interface prevents me from doing so.

This should be fixed now

**Pull request merged**